src/Features/Core/Portable/Diagnostics/DiagnosticAnalyzerService_IncrementalAnalyzer.cs

```diff
@@ -128,6 +129,7 @@ public override void RemoveProject(ProjectId projectId)
             public override bool NeedsReanalysisOnOptionChanged(object sender, OptionChangedEventArgs e)
             {
                 return e.Option.Feature == nameof(SimplificationOptions) ||
+                       e.Option.Feature == nameof(CodeStyleOptions) ||
```

Comment:
and this option is not hidden contract for incremental analyzer

http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/SolutionCrawler/IIncrementalAnalyzer.cs,28

the one who implement incremental analyzer need to explicitly express what option will require it to be re-analyzed.

so diagnostic incremental analyzer also express what require it to re-run. and no diagnostic analyzer itself should have option that affect it. except IDE analyzer decide to do differently.

anyway, we talked about all these and decide to do this way for now until editor config actually implemented all the way down to compiler like rule set file and then we can do same for editor config like rule set file.